rewrite-core/src/main/java/org/openrewrite/internal/ExceptionUtils.java

```diff
@@ -43,4 +46,26 @@ public static String sanitizeStackTrace(Throwable t, Class<?> until) {
         }
         return sanitized.toString();
     }
+
+    public static boolean containsCircularReferences(Throwable exception) {
+        List<Throwable> causes = new LinkedList<Throwable>();
+        causes.add(exception);
+        boolean containsACircularReference = false;
+        while (exception != null && exception.getCause() != null) {
+            Throwable exceptionToFind = exception.getCause();
+            if (exceptionToFind != null) {
+                Optional<Throwable> duplicate = causes.stream().filter(e -> e == exceptionToFind).findFirst();
```

Comment:
With a `Set` as described above we can here use `causes.contains(exceptionToFind)`. Even better:
```java
if (!causes.add(exceptionToFind)) {
    containsACircularReference = true;
    break;
} else {
    exception = exceptionToFind;
}
```